### PR TITLE
[ARGG-1349][documentation]: add deprecated-api, future-api and release-schedule documents

### PR DESCRIPTION
**Pull request by mungodewar**
Adding documentation on new release schedule & the required deprecated api and future api to handle these changes across versions.

Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [x] `README.md` (If you have created a new component)
- [x] Component `README.md`
- [x] Tests
- [x] Accessibility tests
    - The following checks were performed:
        - [ ] Ability to navigate using a [keyboard only](https://webaim.org/techniques/keyboard/)
        - [ ] Zoom functionality ([Deque University explanation](https://dequeuniversity.com/checklists/web/text)):
            - [ ] The page SHOULD be functional AND readable when only the text is magnified to 200% of its initial size
            - [ ] Pages must reflow as zoom increases up to 400% so that content continues to be presented in only one column i.e. Content MUST NOT require scrolling in two directions (both vertically and horizontally)
        - [ ] Ability to navigate using a [screen reader only](https://webaim.org/articles/screenreader_testing/)
- [x] Storybook examples created/updated
- [x] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: decisions/deprecated-api.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Deprecated API
```
Comment (arnaugm) [inline decisions/deprecated-api.md:1]: is `decisions` the right folder for this?

File: decisions/release-schedule.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Release Schedule
+
+## Decision
+Starting Q4 2024, a major version of the @skyscanner/backpack-web package is released approximately every 3 months, typically at the end of each quarter.
+
+### Current Release Schedule
+
+### 2024-2025
```
Comment (arnaugm) [inline decisions/release-schedule.md:8]: do we want to keep this up to date?

File: decisions/future-api.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* Future API **MUST NOT** introduce any breaking change to the default behaviour or appearance of any part of the Stable API
+* There **SHOULD** always be a minimum of 3 months between marking and releasing a feature under a `future` flag and releasing it as the default behaviour in a major version; until then both behaviours should coexist simultaneously.
+
+## How to use?
+* You **SHOULD** aim to opt in to Future API as soon as possible
+
+## Naming Convention
+
+* We use `v{number}__` prefix with two underscores (where applicable) to mark functions, tokens and properties as part of the Future API, and
+* We use `V{number}` prefix to mark React components and props as part of the Future API, where `number` represents the GC library version in which the change will be released as a breaking change.
```
Comment (arnaugm) [inline decisions/future-api.md:24]: needs to be adapted to Backpack

File: decisions/deprecated-api.md (hunk 1 above)
Comment (mungodewar) [reply decisions/deprecated-api.md:1]: I was also a little hesitant, there is not another good place to put documents like this AFAIK and I didn't feel like creating another dir for this. @anambl leaned toward placing these documents here in the ticket.

File: decisions/future-api.md (hunk 3 above)
Comment (mungodewar) [reply decisions/future-api.md:24]: Did multiple passes and still GC refs slip though, nice catch will update.

File: decisions/release-schedule.md (hunk 2 above)
Comment (arnaugm) [reply decisions/release-schedule.md:8]: but this is open source and GC is internal, which may or may not be relevant here, well I guess even if it gets out of date it doesn't matter much as is for illustrative purposes

**Review by arnaugm — APPROVED** [review]

File: decisions/release-schedule.md (hunk 2 above)
Comment (mungodewar) [reply decisions/release-schedule.md:8]: It's hard, requires constant maintenance - I wonder if we should build something into the release process to update these tables?

**Pull request merged**